src/base/ReactorStructs.sol

```diff
@@ -1,22 +1,54 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.0;
 
-import {ERC20} from "solmate/src/tokens/ERC20.sol";
-import {OrderInfo, OutputToken} from "UniswapX/src/base/ReactorStructs.sol";
+import {IReactor} from "UniswapX/src/interfaces/IReactor.sol";
+import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
 
-/// @dev tokens that need to be sent from the swapper in order to satisfy an order
-struct InputTokenWithRecipient {
-    ERC20 token;
-    uint256 amount;
-    // Needed for dutch decaying inputs
-    uint256 maxAmount;
+/// @dev Note that all of these fields are signed over. Some are hashed in the base permit and some are hashed in the passed in witness.
+/// We construct the permit details and witness information.
+struct RelayOrder {
+    // Generic order info
+    OrderInfo info;
+    // Token info for the onchain trade and the payout to fillers
+    Input[] inputs;
+    // The time at which the inputs start decaying
+    uint256 decayStartTime;
+    // The time at which price becomes static
+    uint256 decayEndTime;
+    // ecnoded actions to execute onchain
+    bytes[] actions;
+}
+
+/// @dev generic order information
+///  should be included as the first field in any concrete order types
```

Comment:
I think these comments are all still relevant?

Reply:
Yea I was just suggesting removing "concrete order types", since that refers to the old OrderInfo, which is compatible with normal UniswapX reactors. Important to make the distinction that this local version of OrderInfo is non standard imo